File: Data Structure/Array/Codeforce div2/E_Weird_LCM_Operations.cpp
#include<bits/stdc++.h>
using namespace std;

void solve(int n) {
    vector<pair<int, pair<int, int>>> operations;

    auto add_operation = [&](int i, int j, int k) {
        operations.push_back({i, {j, k}});
    };

    int k = 0;  // Number of operations
    for (int i = n; i >= 2; i--) {
        int j = min(i - 1, int(sqrt(i)) + 1);
        for (; j >= 2; j--) {
            int gcd_ij = __gcd(i, j);
            if (gcd_ij == 1) {
                add_operation(i, j, j - 1);
                k++;
                add_operation(i, j - 1, j - 2);
                k++;
                add_operation(i, j, j - 2);
                k++;
                break;
            }
        }
    }

    // Output the result
    cout << k << '\n';
    for (const auto &op : operations) {
        cout << op.first << ' ' << op.second.first << ' ' << op.second.second << '\n';
    }
}

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;
        solve(n);
    }

    return 0;
}

File: Data Structure/Array/Codeforce div2/E_Weird_LCM_Operations.py
import math

def solve(n):
    operations = []
    
    def add_operation(i, j, k):
        operations.append((i, j, k))
    
    array = [i for i in range(1, n+1)]
    
    def lcm(x, y):
        return x * y // math.gcd(x, y)
    
    for i in range(1, n-1):
        add_operation(i, i+1, i+2)
        array[i-1] = lcm(array[i], array[i+1])
        array[i] = lcm(array[i-1], array[i+1])
        array[i+1] = lcm(array[i-1], array[i])
    
    k = len(operations)
    print(k)
    for op in operations:
        print(*op)

# Read input
t = int(input())
for _ in range(t):
    n = int(input())
    solve(n)
